test/metabase/sync_database/analyze_test.clj

```diff
@@ -72,3 +79,77 @@
 (expect false (values-are-valid-emails? [100]))
 (expect false (values-are-valid-emails? ["true"]))
 (expect false (values-are-valid-emails? ["false"]))
+
+;; Tests to avoid analyzing hidden tables
+(defn count-unanalyzed-fields [db_id table-name]
+  (let [table-id (db/select-one-id Table, :db_id db_id, :active true :name table-name)]
+    (assert (pos? ;; don't let ourselves be fooled if the test passes because the table is
+                  ;; totally broken or has no fields. Make sure we actually test something
+             (db/count Field :table_id table-id)))
+    (db/count Field :last_analyzed nil :table_id table-id)))
+
+(defn latest-sync-time [db_id table-name]
+  (let [table-id (db/select-one-id Table, :db_id db_id, :active true :name table-name)]
+    (db/select-field :last_analyzed Field :table_id table-id)))
```

Comment:
actually on second thought `db/select-field` returns a *set* of values for that field matching the constraints 

```clojure
(db/select-field :last_analyzed 'Field)
;; ->
#{#inst "2017-04-26T21:22:30.523000000-00:00"
  #inst "2017-05-02T23:20:04.154000000-00:00"
  #inst "2017-05-02T22:44:24.734000000-00:00"
  #inst "2017-04-26T21:22:31.025000000-00:00"
  #inst "2017-05-02T23:20:04.273000000-00:00"
  #inst "2017-04-26T21:22:31.159000000-00:00"
  #inst "2017-05-02T22:44:24.652000000-00:00"
  #inst "2017-05-02T23:20:04.128000000-00:00"
  #inst "2017-05-02T22:44:24.672000000-00:00"
  #inst "2017-05-02T22:44:24.775000000-00:00"
  #inst "2017-04-26T21:22:30.727000000-00:00"
  #inst "2017-05-02T23:20:04.229000000-00:00"}
```

I think you might just want to ask it for a single non-nil value of the field

```clojure
(toucan.db/select-one-field :last_analyzed 'Field, :last_analyzed [:not= nil], :table_id <id>)
;; -> #inst "2017-04-26T21:22:31.159000000-00:00"
```